packages/slice-machine/server/src/api/custom-types/save.ts

```diff
@@ -30,13 +33,33 @@ export default async function handler(req: { body: SaveCustomTypeBody }) {
   const modelPath = CustomTypesPaths(env.cwd).customType(model.id).model();
   IO.CustomType.writeCustomType(modelPath, model);
 
-  const mocked = await mock(
+  const libraries = env.manifest.libraries
+    ? Libraries.libraries(env.cwd, env.manifest.libraries)
+    : [];
+
+  const components = libraries.reduce<Component[]>((acc, lib) => {
+    return [...acc, ...lib.components];
+  }, []);
+
+  const sharedSlices = components.reduce<Record<string, SharedSlice>>(
+    (acc, component) => {
+      const slice = Slices.fromSM(component.model);
+      return {
+        ...acc,
+        [slice.id]: slice,
+      };
+    },
+    {}
+  );
+
+  const mocked = mock(
     model,
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-argument
-    CustomTypeMockConfig.getCustomTypeMockConfig(updatedMockConfig, model.id)
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    CustomTypeMockConfig.getCustomTypeMockConfig(updatedMockConfig, model.id),
+    sharedSlices
   );
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-  Files.write(mockPath, mocked as object);
+
+  if (mocked) Files.writeJson(mockPath, mocked);
```

Comment:
this condition should always be defined if using the generate function from Document so no need to check anymore.

Reply:
done